src/tests/EventPipeTracee/CustomMetrics.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using Constants = DotnetCounters.UnitTests.TestConstants;
+
+namespace EventPipeTracee
+{
+    internal sealed class CustomMetrics : IDisposable
+    {
+        private Meter _meter;
+        private Counter<int> _counter;
+        private Histogram<float> _histogram;
+
+        public CustomMetrics()
+        {
+            _meter = new(Constants.TestMeterName);
+            _counter = _meter.CreateCounter<int>(Constants.TestCounter, "dollars");
+            _histogram = _meter.CreateHistogram<float>(Constants.TestHistogram, "feet");
+            // consider adding gauge/etc. here
```

Comment:
Not something we have to do now, but having an ObservableGauge throw an exception during production is a very good test scenario.